### ISSUE TITLE
MMM-MyWastePickup - is it possible to change icons?

### ISSUE THREAD

**Issue by MM2020ops** [jclarke0000/MMM-MyWastePickup#10]
Hi,
I really love your MM-module, especially since it also works with a simple CSV file as input.

But I have a question (problem…).
In our city we have separated collection of Glass and Plastic, so I would really need different icons to display , as the dates can be different .
I could lose the Christmas Tree and replace it with f.i. a Glass Icon , and use the entry in the schedule_custom.csv for the “Christmas” to replace this for “Glass”. In this way there is no code change needed.

I tried to edit/modify/etc… the icon_sprite.svg file myself but without success.

Any possibility to provide an additional icon_xyz.svg file with the XMAS-icon replace with a GLASS-icon of your choice?
Hope you can help… thanks in advance.

**Comment by nitewing76** [comment]
Yes this would be great! Here, christmas tree disposal is part of yard waste...but we have Bulk Trash days where they pick up almost anything. @jclarke0000 please add the ability for custom icons because it would be nice to use [this](https://www.flaticon.com/free-icon/trailer_1747077?term=trailer&page=2&position=96&page=2&position=96&related_id=1747077&origin=search).

**Comment by Imnotme1009** [comment]
Hi,
I'm not sure if this is the right way, but I added some extra icons like this:

- Find the SVG icon you want and name it for example bottle, put it in the MMM-MyWastePickup folder.
- In the MMM-MyWastePickup.js find:

if (pickup.ChristmasTree) {
        iconContainer.appendChild(this.svgIconFactory("christmas_tree"));
      }

- And add:

if (pickup.ChristmasTree) {
        iconContainer.appendChild(this.svgIconFactory("christmas_tree"));
      }
      if (pickup.bottle) {
        iconContainer.appendChild(this.svgIconFactory("bottle"));
      }

- In node_helper.js find:

//reassign strings to booleans for particular waste type
      obj.GreenBin = (obj.GreenBin == "0" ? false : true);
      obj.Garbage = (obj.Garbage == "0" ? false : true);
      obj.Recycling = (obj.Recycling == "0" ? false : true);
      obj.YardWaste = (obj.YardWaste == "0" ? false : true);
      obj.ChristmasTree = (obj.ChristmasTree == "0" ? false : true);
    });

- And add:

//reassign strings to booleans for particular waste type
      obj.GreenBin = (obj.GreenBin == "0" ? false : true);
      obj.Garbage = (obj.Garbage == "0" ? false : true);
      obj.Recycling = (obj.Recycling == "0" ? false : true);
      obj.YardWaste = (obj.YardWaste == "0" ? false : true);
      obj.ChristmasTree = (obj.ChristmasTree == "0" ? false : true);
      obj.bottle = (obj.bottle == "0" ? false : true);
    });

- In /home/mypi4/MagicMirror/css/custom.css add:

.MMM-MyWastePickup .pickup-container .waste-pickup-icon-container .waste-pickup-icon.bottle {
  background-image:url('../modules/MMM-MyWastePickup/bottle.svg');
 }

- Annnnnnd in the schedule_custom.csv you need a new column called bottle

Calendar,WeekStarting,GreenBin,Garbage,Recycling,YardWaste,ChristmasTree,bottle
Custom,03/07/18,1,0,1,0,0,1
Custom,03/14/18,1,1,1,0,0,1
Custom,03/21/18,1,0,1,0,0,1
Custom,03/28/18,1,1,1,0,0,1

Hope this helps
